### ISSUE TITLE
input object fields break usage collection

### ISSUE THREAD

**Issue by rperryng** [charlypoly/graphql-ruby-hive#12]
### Issue:

Given a schema like:

```graphql
type Query {
  projects(filter: FilterInput): [Project!]!
}

input FilterInput {
  type: ProjectType
  pagination: PaginationInput
  order: [ProjectOrderByInput!]
}

input PaginationInput {
  limit: Int
  offset: Int
}

input ProjectOrderByInput {
  field: String!
  direction: OrderDirection
}

enum OrderDirection {
  ASC
  DESC
}

type Project {
  id: ID!
  cleanId: ID!
  name: String!
  type: ProjectType!
  buildUrl: String
  validationUrl: String
}

enum ProjectType {
  FEDERATION
  STITCHING
  SINGLE
  CUSTOM
}
```

and a query such as:
```graphql
query getProjects($limit: Int!, $type: ProjectType!) {
  projects(filter: { pagination: { limit: $limit }, type: $type }) {
    id
  }
}
```

it seems the ruby graphql analyzer API when visiting the `pagination` / `limit` / `type` nodes (i.e. explicit fields of an Input Object):
- calls `on_leave_argument` with a `parent` node without a name method, raising an error when trying to build the field path on [this line](https://github.com/charlypoly/graphql-ruby-hive/blob/master/lib/graphql-hive/analyzer.rb#L18).
- `visitor.argument_definition.type.unwrap.arguments` yields all fields of an input object regardless of whether they were in the query string or not
  - in other words, assuming the error didn't stop execution, the analyzer would return `FilterInput.order` and `PaginationInput.offset` in the used fields result even though it is not present in the query string due to the [logic here](https://github.com/charlypoly/graphql-ruby-hive/blob/master/lib/graphql-hive/analyzer.rb#L24-L26) (I think this is meant to capture all fields in a query where the input object type is referenced directly).

**Comment by rperryng** [comment]
I've started writing some specs for these scenarios [here](https://github.com/charlypoly/graphql-ruby-hive/compare/master...rperryng:graphql-ruby-hive:skip-input-object-fields)

FWIW the schema itself just comes from [JS client tests](https://github.com/kamilkisiela/graphql-hive/blob/9ba16512159bb320c2f4e91486087dd571d8bd7f/packages/libraries/client/tests/usage-collector.spec.ts#L4-L66)

I haven't yet found an obvious/easy solution to these issues with the current implementation:
- One thing to note is that the Visitor implementation in JS offers `ObjectField` visitor hooks which is how the `@graphql-hive/client` implementation deals with these fields ([code](https://github.com/kamilkisiela/graphql-hive/blob/9ba16512159bb320c2f4e91486087dd571d8bd7f/packages/libraries/client/src/internal/usage.ts#L389)).  The Ruby GraphQL's visitor doesn't offer these hooks
- GraphQL Ruby offer `visitor.query.arguments_for(node, visitor.field_definition)` ([related docs](https://graphql-ruby.org/queries/ast_analysis.html#field-arguments)).  I need to look more deeply, but I think it does provide the input object "structure" to manually traverse through, but in other simpler cases didn't return _any_ arguments...
- `node.arguments` during the `on_leave_field` hook might also offer something similar

**Comment by rperryng** [comment]
I have also raised an issue in the `graphql-ruby` repo (https://github.com/rmosolgo/graphql-ruby/issues/4332) to ask for assistance.

**Comment by rperryng** [comment]
I think I've figured it out in https://github.com/charlypoly/graphql-ruby-hive/pull/13, and now that I've gotten the nice solution it seems so obvious and simple in hindsight.  🥴.
